### ISSUE TITLE
log10 transformation of y-axis values in marginal plot of ggscatterhist() function 

### ISSUE THREAD

**Issue by marcomeola** [kassambara/ggpubr#358]
Is it possible to transform the y-axis of the density plot corresponding to Sepal.Length as log10?

Minimal example:
plot <- ggscatterhist(
  iris, x = "Sepal.Length", y = "Sepal.Width",
  color = "Species", size = 3, alpha = 0.6,
  margin.params = list(fill = "Species", color = "black", size = 0.2),
  margin.plot="density"
  )

**Comment by kassambara** [comment]
Yes — the mechanism is to ask `ggscatterhist()` for the pieces instead of a finished plot: with `print = FALSE` it returns a list of the three sub-plots (`sp` = scatter, `xplot` = the **Sepal.Length** marginal, `yplot` = the Sepal.Width marginal). You can then edit any of them like a normal `ggplot` and print:

```r
plots <- ggscatterhist(
  iris, x = "Sepal.Length", y = "Sepal.Width", color = "Species",
  size = 3, alpha = 0.6, margin.plot = "density",
  margin.params = list(fill = "Species", color = "black", size = 0.2),
  print = FALSE)

plots$xplot <- plots$xplot + scale_y_log10()   # y of the top marginal = density
plots
```

One caveat, so I can point you at the right thing: the **y-axis of that density plot is the density value**, and log10 of a density is awkward — densities are often < 1 (so the log is negative) and touch 0 (→ `-Inf`), which makes the curve collapse; ggscatterhist also hides the marginal axis, so the change isn't even visible. So that's rarely what people actually want.

If instead you meant putting the **Sepal.Length values** on a log scale (log the measurement, not the density), apply `scale_x_log10()` to both the scatter and the top marginal so they stay aligned:

```r
plots$sp    <- plots$sp    + scale_x_log10()
plots$xplot <- plots$xplot + scale_x_log10()
plots
```

Which of the two did you have in mind? Happy to give a complete, tidy example once I know — leaving this open so you can confirm.

**Comment by kassambara** [comment]
Yes — this is possible. `ggscatterhist()` returns a **list of three ggplot objects** that you can modify before printing:

- `$sp` — the main scatter plot
- `$xplot` — the top marginal (for the `x` variable, here `Sepal.Length`)
- `$yplot` — the right marginal (for the `y` variable, here `Sepal.Width`)

So you add any ggplot2 scale (including a log transform) to the relevant component(s). The only rule is: to keep a marginal aligned with the scatter, apply the **same** transform to the scatter and to that marginal.

**Put `Sepal.Length` on a log10 scale** (it's the `x` variable → transform `$sp` and the top marginal `$xplot`):

``` r
library(ggpubr)
plot <- ggscatterhist(
  iris, x = "Sepal.Length", y = "Sepal.Width",
  color = "Species", size = 3, alpha = 0.6,
  margin.params = list(fill = "Species", color = "black", size = 0.2),
  margin.plot = "density"
)
plot$sp    <- plot$sp    + scale_x_log10()
plot$xplot <- plot$xplot + scale_x_log10()
plot
```

![](https://i.imgur.com/5Le0aEN.png)

Note the compressed `5 → 6 → 7 → 8` spacing on the x-axis, and that the top density panel stays aligned.

A few variants, depending on which axis you actually mean:

- **`Sepal.Width` (the main plot's y-axis) on log10** → transform `$sp` and the right marginal instead:
  `plot$sp <- plot$sp + scale_y_log10(); plot$yplot <- plot$yplot + scale_x_log10()`
  (the right marginal is rotated, so its `Sepal.Width` values live on *its* x-axis).
- **The density values themselves on log10** (the top panel's own y-axis) → `plot$xplot <- plot$xplot + scale_y_log10()`; this one doesn't need to match the scatter.

Hope that covers your case — let me know which axis you meant if not, and I'll adjust. Closing as resolved.
